README.md

```diff
@@ -2,8 +2,147 @@
 
 [![Build Status](https://travis-ci.org/hmrc/file-transmission.svg)](https://travis-ci.org/hmrc/file-transmission) [ ![Download](https://api.bintray.com/packages/hmrc/releases/file-transmission/images/download.svg) ](https://bintray.com/hmrc/releases/file-transmission/_latestVersion)
 
-This is a placeholder README.md for a new repository
+Microservice that facilitates transmission of files between MDTP and MDG.
+
+Whenever service on MDTP platform wants to send bundle (batch) of files to MDG it will need to use this service. The file-transmission
+service notifies MDG that files prepared by service on MDTP are ready to be processed. Please note that file-transmission service 
+do not transfer file contents. The service just passes data about file location. The file content has to be provided by other service (eg. upscan)
+
+# Typical usage scenario
+
+- Consuming service requests upload of the file from user using `upscan` service
+- `Upscan` notifies consuming service that the file has been uploaded and is ready to be processed - giving consuming 
+service the URL that allows to download the file.
+- Consuming service ensures that all required files have been uploaded by the user
+- Consuming service uses file-transmission service to notify MDG that files are ready to be processes
+- MDG asynchronously uses the file
+
+## Service usage
+
+### Whitelisting
+
+In order to initiate an upload the consuming service must be whitelisted by file-transmission. See the 'Whitelisting client services' section further down in this document.
+
+### Requesting a file transmission
+
+The basic unit of work for file-transmission service is a batch that might consist of one or more files.
+
+Information about every file in the batch is passed to the file-transmission service as a separate POST request. 
+There is no need to make additional call to create a batch or to notify that data about all files in the batch have 
```

Comment:
I think this part is a bit confusing, but might call into question the business requirements for batch here. I wonder whether it makes more sense that a single post request can contain multiple files and the same batch, rather than needing separate calls, otherwise it almost makes the point of having more than one file per batch redundant. Question for the business probably.